diff -urN --no-dereference mlibc-clean/meson.build mlibc-workdir/meson.build
--- mlibc-clean/meson.build
+++ mlibc-workdir/meson.build
@@ -273,6 +273,10 @@
 	rtld_include_dirs += include_directories('sysdeps/nyaux/include')
 	libc_include_dirs += include_directories('sysdeps/nyaux/include')
 	subdir('sysdeps/nyaux')
+elif host_machine.system() == 'willowos'
+	rtld_include_dirs += include_directories('sysdeps/willowos/include')
+	libc_include_dirs += include_directories('sysdeps/willowos/include')
+	subdir('sysdeps/willowos')
 else
 	error('No sysdeps defined for OS: ' + host_machine.system())
 endif
